schema-registry/src/main/java/io/streamnative/pulsar/handlers/kop/schemaregistry/resources/SchemaResource.java

```diff
@@ -40,15 +41,23 @@ public SchemaResource(SchemaStorageAccessor schemaStorageAccessor,
      */
     public void register(SchemaRegistryHandler schemaRegistryHandler) {
         schemaRegistryHandler.addProcessor(new GetSchemaById());
+        schemaRegistryHandler.addProcessor(new GetSchemaStringById());
         schemaRegistryHandler.addProcessor(new GetSchemaTypes());
         schemaRegistryHandler.addProcessor(new GetSchemaAliases());
     }
 
     @Data
     @NoArgsConstructor
     @AllArgsConstructor
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
     public static final class GetSchemaResponse {
         private String schema;
+        private String schemaType;
+        //todo: references, metadata, ruleSet, maxId
+
+        public String getSchemaType() {
+            return "AVRO".equals(schemaType) ? null : schemaType;
+        }
```

Comment:
Currently KoP does not support these fields, e.g. the references from https://github.com/confluentinc/schema-registry/pull/1280. Actually these fields were introduced in some latest versions of the Confluent Schema Registry. 

We can open an issue to track these features. But for now, this code is meaningless.

Reply:
Good to know. So for confluent schema, it only supports AVRO schema by default? If we want to use Protobuf or JSON schema, do we need to add different schema.providers?

IMO, if confluent's schema only returns schema type field for non-avro schemas. I think it is meaningful or just good to have (in the end, we will have it.)

Reply:
No. It supports all of Avro, JSON, Protobuf.

I mean, the schema type field is only related to the [schema references feature](https://akhq.io/docs/configuration/schema-registry/schema-references.html), which we don't support yet. The schema type field here is added only for the "references" field in your TODO comment. So if you added a schema type field but didn't support schema references, the code here would be meaningless.

Reply:
> This feature is less important for Avro since the feature is associated with their protocol functionality, which is not used by Schema Registry.

I see the schema reference feature is mainly for JSON and Protobuf schemas.

So it's okay to have `getSchemaType` here. The type field might be passed from the Confluent's Avro serializer. In this case, we might return "type: JSON" or "type: Protobuf" to users.